library/network_connections.py

```diff
@@ -1061,6 +1061,56 @@ def volatilize_connection(self, connection, timeout=10):
                 % (cb_args.get("error", "unknown error"), cb_args)
             )
 
+    def create_checkpoint(self):
+        """ Create a new checkpoint """
+        checkpoint = self._call_checkpoint(
+            "create",
+            [],  # devices, empty list is all devices
+            60,  # timeout in seconds, FIXME: Make it an option
+            Util.NM().CheckpointCreateFlags.DELETE_NEW_CONNECTIONS
+            | Util.NM().CheckpointCreateFlags.DISCONNECT_NEW_DEVICES,
+        )
+
+        if checkpoint:
+            return checkpoint.get_path()
+
+    def destroy_checkpoint(self, path):
+        """ Destroy the specified checkpoint """
+        self._call_checkpoint("destroy", path)
+
+    def rollback_checkpoint(self, path):
+        """ Rollback the specified checkpoint """
+        self._call_checkpoint("rollback", path)
+
+    def _call_checkpoint(self, action, *args):
+        action_name = "checkpoint_" + action
+        finish = action_name + "_finish"
+        callback = Util.create_callback(finish)
+        cancellable = Util.create_cancellable()
+        user_data = {}
+
+        mainloop_timeout = 10
```

Comment:
timeout of 10 might not be enough for rollback.

Reply:
What is your recommendation for the timeout?

Reply:
it probable should be the same as it takes to activate one profile... see
```
            wait_time = connection["wait"]
            if wait_time is None:
                wait_time = 90

            try:
                self.nmutil.connection_activate_wait(ac, wait_time)
```

Reply:
changed it to 60 for the rolback.

Reply:
Why do you like that better?

Reply:
I did not notice your comment when choosing 60, I will change it to 90.